frodo.js

```diff
@@ -1,4 +1,4 @@
-const uniqueValues = [...new Set(array)];
```

Comment:
SCRIPT-The way you've refactored this complex method into simpler ones is great. It makes the code much more maintainable.